src/OpenCV/ImgProc/GeometricImgTransform.hsc

```diff
@@ -297,3 +298,37 @@ getRotationMatrix2D center angle scale = unsafeCoerceMat $ unsafePerformIO $
   where
     c'angle = realToFrac angle
     c'scale = realToFrac scale
+
+{- | Applies a generic geometrical transformation to an image.
+
+<http://docs.opencv.org/3.0-last-rst/modules/imgproc/doc/geometric_transformations.html#remap>
+-}
+remap
+    :: Mat ('S [inputHeight, inputWidth]) inputChannels inputDepth
+    -> Mat ('S [outputHeight, outputWidth]) ('S 2) ('S Float)
+    -> InterpolationMethod
+    -> BorderMode
+    -> Mat ('S [outputHeight, outputWidth]) inputChannels inputDepth
+remap src mapping interpolationMethod borderMode =
+  unsafeCoerceMat $ unsafePerformIO $ do
+    dst <- newEmptyMat
+    withPtr src $ \srcPtr ->
+      withPtr dst $ \dstPtr ->
+        withPtr mapping $ \mappingPtr ->
+          withPtr borderValue $ \borderValuePtr ->
+            [CU.block| void {
+              Mat * src = $(Mat * srcPtr);
+              cv::remap
+                ( *src
```

Comment:
If you only need `srcPtr` once in the splices code you may directly place the splice where needed.

So `*$(Mat * srcPtr)` instead of `*src`. Then you can also remove the local binding.